optuna/storages/journal/_file.py

```diff
@@ -171,12 +173,20 @@ def acquire(self) -> bool:
                                 "The existing lock file has not been released "
                                 "for an extended period. Forcibly releasing the lock file."
                             )
+                            last_warning_time = time.monotonic()
                             try:
                                 self.release()
                                 sleep_secs = 0.001
                             except RuntimeError:
                                 continue
 
+                        if time.monotonic() - last_warning_time > warning_interval:
+                            optuna_warn(
+                                f"It is taking longer than {warning_interval} seconds to acquire "
+                                f"the lock file: {self._lock_file} Retrying..."
+                            )
+                            last_warning_time = time.monotonic()
```

Comment:
How about issuing a warning even when `self.grace_period` is `None`?

```suggestion
                    if time.monotonic() - last_warning_time > warning_interval:
                        optuna_warn(
                            f"It is taking longer than {warning_interval} seconds to acquire "
                            f"the lock file: {self._lock_file} Retrying..."
                        )
                        last_warning_time = time.monotonic()
```